platform/mbed_application.c

```diff
@@ -48,12 +48,20 @@ static void powerdown_nvic()
     int i;
     int j;
 
+#if defined(__CORTEX_M23)
+    isr_groups_32 = 16;
```

Comment:
16 is not a good number... and more than even a M33 could support.  The max number of external interrupts a M23 can support is 240.  It's sad that M23 doesn't have the ICTR so the max value we could use here is 7 (or 8... depending on how it's used elsewhere).  Rule SGCR in v8-M ARM states:  "When a particular NVIC interrupt line is not implemented, the registers that are associated with it are reserved."  Yet Rule GLNG states "Privileged accesses to unimplemented registers are RES0" so accessing non-existing registers should be harmless but are wasteful.

Reply:
@TacoGrandeTX Thanks for your detailed information. I fix it in https://github.com/ARMmbed/mbed-os/pull/6949/commits/3c14cb633222cbc42fc9c1ea257da49db3a73d15.